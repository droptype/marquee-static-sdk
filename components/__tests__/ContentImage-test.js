const React = require('react')
const ContentImage = require('../ContentImage')
const renderer = require('react-test-renderer')

const SAMPLE_BLOCK = require('../../entry/__tests__/SAMPLE_ENTRY.json').content.find( block => 'image' === block.type)

describe('ContentImage', () => {
    it('renders', () => {
        const component = renderer.create(
            React.createElement(ContentImage, { src: SAMPLE_BLOCK })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
    it('renders plain', () => {
        const component = renderer.create(
            React.createElement(ContentImage, { src: SAMPLE_BLOCK, plain: true })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
})
