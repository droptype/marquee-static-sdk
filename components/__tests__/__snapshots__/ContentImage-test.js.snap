// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentImage renders 1`] = `
<img
  alt="The spire of the Crysler building in the background, the tops of various skyscrapers in front of it."
  className={
    Classes {
      "_groups": Object {},
      "_initial": Array [
        "ContentImage",
      ],
      "_switches": Object {},
    }
  }
  data-src_128="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg"
  data-src_1280="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg"
  data-src_2560="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg"
  data-src_640="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
  sizes="(min-width: 1440px) 1440px,(min-width: 720px) 720px,(min-width: 360px) 360px,100vw"
  src="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
  srcSet="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg 128w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg 640w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg 1280w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg 2560w"
/>
`;

exports[`ContentImage renders plain 1`] = `
<img
  src="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
/>
`;
